type-check.ts

```diff
@@ -313,16 +392,67 @@ export function tcExpr(env : GlobalTypeEnv, locals : LocalTypeEnv, expr : Expr<n
       } else {
         throw new TypeError("Undefined function: " + expr.name);
       }
+    case "list-comp":
+      // check if iterable is instance of Range class
+      const iterable = tcExpr(env, locals, expr.iterable);
+      if (iterable.a.tag === "class" && iterable.a.name === "range"){
```

Comment:
I would definitely generalize this to things other than `range`. We suggested considering any class that implements a `next` and `hasnext` as a possible interface.

Reply:
We plan on fixing this hardcoding as part of our next milestone. Please let us know if that is not okay, we will try to resolve this.